shell/src/main/java/alluxio/cli/ValidateEnv.java

```diff
@@ -0,0 +1,266 @@
+        node, remoteCommand);
+    String[] command = {"bash", "-c", localCommand};
+    try {
+      ProcessBuilder builder = new ProcessBuilder(command);
+      builder.redirectErrorStream(true);
+      builder.redirectOutput(ProcessBuilder.Redirect.INHERIT);
+      builder.redirectInput(ProcessBuilder.Redirect.INHERIT);
+      Process process = builder.start();
+      process.waitFor();
+      return process.exitValue() == 0;
+    } catch (IOException e) {
+      System.err.format("Unable to validate on node %s: %s.%n", node, e.getMessage());
+      return false;
+    }
+  }
+
+  // runs validation tasks in local environment
+  private static boolean validateLocal(String target, String name) {
+    int validationCount = 0;
+    int failureCount = 0;
+    Collection<ValidationTask> tasks = TARGET_TASKS.get(target);
+    for (ValidationTask task : tasks) {
+      String taskName = TASKS.get(task);
+      if (name != null && !taskName.startsWith(name)) {
+        continue;
+      }
+
+      System.out.format("Validating %s...", taskName);
+      if (task.validate()) {
+        System.out.println("OK");
+      } else {
+        System.out.println("Failed");
+        failureCount++;
+      }
+
+      validationCount++;
+    }
+
+    if (failureCount > 0) {
+      System.err.format("Validation failed. Total failures: %d.%n", failureCount);
+      return false;
+    }
+
+    if (validationCount == 0) {
+      System.err.format("No validation task matched name \"%s\".%n", name);
+      return false;
+    }
+
+    System.out.println("Validation succeeded.");
+    return false;
+  }
+
+  private static boolean validateWorkers(String name) throws InterruptedException {
+    return validateRemote(Utils.readNodeList("workers"), "worker", name);
+  }
+
+  private static boolean validateMasters(String name) throws InterruptedException {
+    return validateRemote(Utils.readNodeList("masters"), "master", name);
+  }
+
+  /**
+   * Prints the help message.
+   *
+   * @param message message before standard usage information
+   */
+  public static void printHelp(String message) {
+    System.err.println(message);
+    System.out.print(USAGE);
+  }
+
+  /**
+   * Validates environment.
+   *
+   * @param args list of arguments
+   * @return 0 on success, -1 on validation failures, -2 on invalid arguments
+   */
+  public static int validate(String... args) throws InterruptedException {
+    if (args.length < 1) {
+      printHelp("Target not specified.");
+      return -2;
+    }
+
+    if (args.length > 2) {
+      printHelp("More arguments than expected.");
+      return -2;
+    }
+
+    String target = args[0];
+    String name = args.length > 1 ? args[1] : null;
+
+    boolean success;
+    switch (target) {
+      case "local":
+      case "worker":
+      case "master":
+        success = validateLocal(target, name);
+        break;
+      case "all":
+        success = validateMasters(name);
+        success = validateWorkers(name) && success;
```

Comment:
I broke it down to two lines because the operator `&&` expression will not evaluate the second part if the first part is evaluated as false.